colossalai/inference/kv_cache/kvcache_manager.py

```diff
@@ -223,46 +223,58 @@ def allocate_context_from_block_tables(self, block_tables: torch.Tensor, context
             self._block_states_cum[:-num_blocks_required],
             out=self._block_finder[num_blocks_required - 1 :],
         )
+
         end_indexes = torch.nonzero(self._block_finder == num_blocks_required, as_tuple=False).view(-1)
+
+        block_tables_list = block_tables.tolist()
+        blocks_required_list = blocks_required.tolist()
+
         if end_indexes.numel() > 0:
             # contiguous cache exists
             end_idx = end_indexes[0].item() + 1  # open interval
             start_idx = end_idx - num_blocks_required  # closed interval
-            alloc_block_ids = torch.arange(start_idx, end_idx)
+            alloc_block_ids = torch.arange(start_idx, end_idx, device=block_tables.device)
+
             for i in range(bsz):
-                curr_required = blocks_required[i]
-                block_tables[i, :curr_required] = torch.arange(
-                    start_idx, start_idx + curr_required, device=block_tables.device
-                )
+                curr_required = blocks_required_list[i]
+                for j in range(curr_required):
+                    block_tables_list[i][j] = start_idx + j
                 start_idx += curr_required
         else:
             # non-contiguous cache
             available_block_ids = torch.nonzero(self._block_states > 0).view(-1)
             alloc_block_ids = available_block_ids[:num_blocks_required]
             alloc_block_ids = alloc_block_ids.to(dtype=block_tables.dtype, device=block_tables.device)
+            alloc_block_ids_list = alloc_block_ids.tolist()
```

Comment:
This 'tolist' only functions under “non-contiguous cache” and is used to optimize the processing logic of “non-contiguous cache”. Since the alloc_block_ids tensor will still be used later, we may not convert it to a list ahead of time

Reply:
Yes. Just looking for ways to make code cleaner and less redundant. It's fine here.